src/main/java/uk/gov/gchq/hqdm/services/SpatioTemporalExtentServices.java

```diff
@@ -0,0 +1,1193 @@
+package uk.gov.gchq.hqdm.services;
+
+import uk.gov.gchq.hqdm.iri.HQDM;
+import uk.gov.gchq.hqdm.iri.IRI;
+import uk.gov.gchq.hqdm.iri.IriBase;
+import uk.gov.gchq.hqdm.model.*;
+import uk.gov.gchq.hqdm.model.impl.*;
+import uk.gov.gchq.hqdm.util.UID;
+
+/**
+ * Utilities for building and generating HQDM objects.
+ */
+public class SpatioTemporalExtentServices {
+    /**
+     * Generate a new PointInTime.
+     *
+     * @param eventTime Entity name of the point in time.
+     * @param pw The PossibleWorld the PointInTime is a part of.
+     * @param baseIri IriBase of the PointInTime.
+     * @return The generated PointInTime.
+     */
+    public static PointInTime event(final String eventTime, final PossibleWorld pw,
+            final IriBase baseIri) {
+        final PointInTime timeEvent = new PointInTimeImpl(new IRI(baseIri, UID.uid()));
```

Comment:
Would it be more logical for this function to receive an existing `PointInTime` which is then assigned the possible world association, rather than returning a new instance of `PointInTime` with the added possible world association? That way this method can be used with points in time that have already been defined using the `createPointInTime()` method. This would also keep this method pure by avoiding creating *and* manipulating the `timeEvent` before returning it. At which point I wonder if this function is strictly necessary, as adding the `ENTITIY_NAME` and `PART_OF_POSSIBLE_WORLD` associations is something the user could do manually?

Reply:
This is a reproduction of the convenience function `DataObjectUtils.event()` from `MagmaCore` - it is used by the `ExampleDataObjects` class and it becomes redundant in when the examples are updated, so I will delete it now. There will be a better `ExampleDataObjects` in an up-coming `MagmaCore` PR that does not depend on this function.